modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/snapshot/MajorityPeerSelection.scala

```diff
@@ -0,0 +1,60 @@
+package org.tessellation.sdk.infrastructure.snapshot
+
+import cats.data.NonEmptyList
+import cats.effect.kernel.Async
+import cats.effect.std.Random
+import cats.syntax.all._
+
+import org.tessellation.kryo.KryoSerializer
+import org.tessellation.schema.node.NodeState.Ready
+import org.tessellation.schema.peer.Peer
+import org.tessellation.schema.peer.Peer.toP2PContext
+import org.tessellation.sdk.domain.cluster.storage.ClusterStorage
+import org.tessellation.sdk.domain.snapshot.PeerSelection
+import org.tessellation.sdk.http.p2p.clients.L0GlobalSnapshotClient
+import scala.util.control.NoStackTrace
+import cats.ApplicativeError
+
+object MajorityPeerSelection {
+
+  case object NoPeers extends NoStackTrace
+
+  def make[F[_]: Async: KryoSerializer: Random](
+    storage: ClusterStorage[F],
+    snapshotClient: L0GlobalSnapshotClient[F]
+  ): PeerSelection[F] = new PeerSelection[F] {
+
+    def select: F[Peer] = storage.getResponsivePeers
+      .map(_.filter(_.state === Ready))
+      .flatMap(getPeerSublist)
+      .flatMap(peers => ApplicativeError[F, Throwable].fromOption(peers.toNel, NoPeers))
+      .flatMap(filterPeerList)
+      .flatMap(filteredPeers => Random[F].elementOf(filteredPeers.toList))
```

Comment:
it can be simplified to `map(_.toList).flatMap(Random[F].elementOf)`

Reply:
Changed.